build_defs/cpp_linenoise.BUILD

```diff
@@ -0,0 +1,20 @@
+load("@rules_cc//cc:defs.bzl","cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
+exports_files(["LICENSE"])
+
+COPTS = select({
+    "//conditions:default": [
+        "-fno-exceptions",
+        "-fno-rtti",
+    ],
+})
+
+cc_library(
+    name = "cpp_linenoise",
+    hdrs = ["linenoise.hpp"],
+    includes = ["."],
+)
```

Comment:
Let's add `include_prefix="cpp_linenoise"` to make it more "contained" 
https://docs.bazel.build/versions/master/be/c-cpp.html#cc_library.include_prefix

Then in the code it will be `#include "cpp_linenoise/linenoise.hpp"`.